exercises/go-counting/src/test/java/GoCountingTest.java

```diff
@@ -0,0 +1,178 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.awt.Point;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+public class GoCountingTest {
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none(); 
+	
+	String board5x5 = "  B  \n" +
+	                  " B B \n" +
+	                  "B W B\n" +
+	                  " W W \n" +
+	                  "  W  ";
+	
+	@Test
+	public void blackCorner5x5BoardTest() {
+		GoCounting gocounting = new GoCounting(board5x5);
+		
+		Set<Point> territory = new HashSet<>();
+		territory.add(new Point(0, 0));
+		territory.add(new Point(0, 1));
+		territory.add(new Point(1, 0));
+		
+		assertEquals(gocounting.BLACK, gocounting.getTerritoryOwner(0, 1));
+		assertEquals(territory, gocounting.getTerritory(0, 1));
```

Comment:
Does this give a good error mesage when the test fails? That is, does it point out the exact difference between the two sets?

Reply:
It seems to. For example if I remove a point from the expected set I get:

GoCountingTest > blackCorner5x5BoardTest FAILED
    java.lang.AssertionError: expected:<[java.awt.Point[x=0,y=0], java.awt.Point[x=0,y=1]]> but was:<[java.awt.Point[x=0,y=1], java.awt.Point[x=0,y=0], java.awt.Point[x=1,y=0]]>